src/certificate.rs

```diff
@@ -40,8 +41,12 @@ pub struct Certificate {
 
 #[derive(ThisError, Debug)]
 pub enum CertificateError {
-    #[error("Error while decoding a certificate: {0}")]
-    DecodeError(String),
+    #[error("Incorrect format of the certificate: {0}")]
+    IncorrectCertificateFormat(String),
+    #[error("Incorrect length of an array. Should be 2 bytes of a format, 4 bytes of a version and 104 bytes for each trust")]
+    IncorrectByteLength,
+    #[error("Error while decoding a trust in a certificate: {0}")]
+    DecodeError(TrustError),
```

Comment:
Can you just do?
```suggestion
    DecodeError(#[from]TrustError),
```

and maybe also #[source], not sure. That way, you can drop `map_err(DecodeError)` and just use `?`